server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/utils/ParametersExtractor.java

```diff
@@ -0,0 +1,92 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.webadmin.utils;
+
+import java.util.Optional;
+
+import org.apache.james.util.streams.Limit;
+import org.apache.james.util.streams.Offset;
+import org.eclipse.jetty.http.HttpStatus;
+
+import com.google.common.base.Strings;
+
+import spark.Request;
+
+public class ParametersExtractor {
+
+    public static final String LIMIT_PARAMETER_NAME = "limit";
+    public static final String OFFSET_PARAMETER_NAME = "offset";
+
+    public static Limit extractLimit(Request request) {
+        return Limit.from(assertPositiveInteger(request, LIMIT_PARAMETER_NAME)
+                .map(value -> assertNotZero(value, LIMIT_PARAMETER_NAME)));
+    }
+
+    public static Offset extractOffset(Request request) {
+        return Offset.from(assertPositiveInteger(request, OFFSET_PARAMETER_NAME));
+    }
+
+    public static Optional<Double> extractPositiveDouble(Request request, String parameterName) {
+        return extractPositiveNumber(request, parameterName)
+                .map(Number::doubleValue);
+    }
+
+    private static Optional<Integer> assertPositiveInteger(Request request, String parameterName) {
+        return extractPositiveNumber(request, parameterName)
```

Comment:
I missed this call on previous review. I doesn't make sense to parse as Double than convert to an int.
I suggestion changing extractPositiveNumber to : 
```private static <T extends Number> Optional<T> extractPositiveNumber(Request request, String parameterName, Function<String, T> toNumber) ```
That way, ```assertPositiveInteger``` will only do ```extractPositiveNumber(request, parameterName, Integer::valueOf)``` and ```assertPositiveDouble``` will do ```extractPositiveNumber(request, parameterName, Double::valueOf)```
WDYT ?

Reply:
See XXX